src/content/components/NodeIcon.js

```diff
@@ -0,0 +1,35 @@
+import React, { PureComponent, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import { getIconClassNameForNode } from '../reducers/icons';
+import actions from '../actions';
+
+class NodeIcon extends PureComponent {
+  constructor(props) {
+    super(props);
+    if (props.icon) {
+      props.iconStartLoading(props.icon);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.icon) {
+      nextProps.iconStartLoading(nextProps.icon);
+    }
+  }
+
+  render() {
+    return <div className={`treeRowIcon ${this.props.className || ''}`}></div>;
```

Comment:
I think it's more useful if we have a boolean to decide whether to include a class; but this could still work in this case, up to you :)